tmail-backend/tmail-third-party/ai-bot/src/test/java/com/linagora/tmail/mailet/rag/EmailParserTest.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.StoreMailboxManager;
+import org.apache.james.metrics.tests.RecordingMetricFactory;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.message.DefaultMessageBuilder;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.ClassLoaderUtils;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.james.utils.UpdatableTickingClock;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+
+public class EmailParserTest {
+    private static final Username BOB = Username.of("bob@test.com");
+    private static final MailboxPath BOB_INBOX_PATH = MailboxPath.inbox(BOB);
+
+    MessageManager bobInboxMessageManager;
+    StoreMailboxManager mailboxManager;
+    UpdatableTickingClock clock;
+    MailboxSessionMapperFactory mapperFactory;
+    MemoryEventDeadLetters eventDeadLetters;
+    MailboxSession bobMailboxSession;
+    MailboxId bobInboxId;
+
+    private static final byte[] CONTENT = (
+        "Subject: Test Subject\r\n" +
+            "From: sender@example.com\r\n" +
+            "To: recipient@example.com\r\n" +
+            "Cc: cc@example.com\r\n" +
+            "Date: Tue, 10 Oct 2023 10:00:00 +0000\r\n" +
+            "\r\n" +
+            "Body of the email").getBytes(StandardCharsets.UTF_8);
+
+    @BeforeEach
+    void setUp() throws MailboxException {
+        clock = new UpdatableTickingClock(Instant.now());
+        mapperFactory = new InMemoryMailboxSessionMapperFactory(clock);
+        RetryBackoffConfiguration backoffConfiguration = RetryBackoffConfiguration.builder()
+            .maxRetries(2)
+            .firstBackoff(Duration.ofMillis(1))
+            .jitterFactor(0.5)
+            .build();
+        eventDeadLetters = new MemoryEventDeadLetters();
+        InMemoryIntegrationResources resources = InMemoryIntegrationResources.builder()
+            .preProvisionnedFakeAuthenticator()
+            .fakeAuthorizator()
+            .eventBus(new InVMEventBus(new InVmEventDelivery(new RecordingMetricFactory()), backoffConfiguration, eventDeadLetters))
+            .defaultAnnotationLimits()
+            .defaultMessageParser()
+            .scanningSearchIndex()
+            .noPreDeletionHooks()
+            .storeQuotaManager()
+            .build();
+        bobMailboxSession = MailboxSessionUtil.create(BOB);
+        mailboxManager = resources.getMailboxManager();
+        bobInboxId = mailboxManager.createMailbox(BOB_INBOX_PATH, bobMailboxSession).get();
+        bobInboxMessageManager = mailboxManager.getMailbox(bobInboxId, bobMailboxSession);
+    }
```

Comment:
ZERO setup is expected to test a class as easy as `EmailParser`.

I'm expecting perfect unit tests: text in, assertions on test out.

If we wish to write more complete integration tests, then let's do it into another class.